src/v2/Apps/Conversation/Components/Message.tsx

```diff
@@ -1,42 +1,83 @@
 import {
   Box,
   BoxProps,
+  Color,
+  DownloadIcon,
   Flex,
+  FlexProps,
   Image,
-  Link,
   Sans,
-  Serif,
   Spacer,
+  color,
 } from "@artsy/palette"
 import { Message_message } from "v2/__generated__/Message_message.graphql"
 import React from "react"
 import { createFragmentContainer } from "react-relay"
 import { graphql } from "relay-runtime"
 import { TimeSince } from "./TimeSince"
+import styled from "styled-components"
+import {
+  AlignSelfProps,
+  BackgroundProps,
+  alignSelf,
+  background,
+} from "styled-system"
+
+const AttachmentLink = styled.a`
+  width: min-content;
+  text-decoration: none;
+  max-width: 66.67%;
+`
+
+const AttachmentContainer = styled(Flex)<
+  FlexProps & AlignSelfProps & BackgroundProps
+>`
+  ${alignSelf};
+  ${background};
+  border-radius: 15px;
+  white-space: no-wrap;
+  width: min-content;
+  justify-content: space-between;
+`
 
 interface AttachmentProps {
-  item: Message_message["attachments"][0]
+  attachment: Message_message["attachments"][0]
+  alignSelf: string
+  bgColor: Color
+  textColor: Color
 }
 
 export const Attachment: React.FC<AttachmentProps> = props => {
-  const { item } = props
-  if (item.contentType.startsWith("image")) {
+  const { attachment, alignSelf, bgColor, textColor } = props
+  if (attachment.contentType.startsWith("image")) {
     return (
-      <Flex flexDirection="column" m={2}>
-        <Image src={item.downloadURL} width="75px" title={item.fileName} />
-        <Link href={item.downloadURL}>
-          <Serif size="2">{item.fileName}</Serif>
-        </Link>
-      </Flex>
+      <AttachmentLink href={attachment.downloadURL} target="_blank">
+        <AttachmentContainer
+          p={1}
+          mt={0.5}
+          alignSelf={alignSelf}
+          background={color(bgColor)}
+        >
+          <Image src={attachment.downloadURL} />
+        </AttachmentContainer>
+      </AttachmentLink>
     )
-  } else if (item.contentType === "application/pdf") {
+  } else {
     return (
-      <Box>
-        <Link href={item.downloadURL}>{item.fileName}</Link>
-      </Box>
+      <AttachmentLink href={attachment.downloadURL} target="_blank">
+        <AttachmentContainer
+          alignSelf={alignSelf}
+          background={color(bgColor)}
+          p={1}
+          mt={0.5}
+        >
```

Comment:
#veryminor seems possible to move the `if/else` clause inside `AttachementContainer` to make the code a little bit dryer.

Reply:
That's a good catch!